src/runner/settings_window.cpp

```diff
@@ -183,66 +188,93 @@ void run_settings_window()
     wcscat_s(executable_args, std::to_wstring(powertoys_pid).c_str());
     wcscat_s(executable_args, settings_theme.c_str());
 
-    // Run the Settings process with non-elevated privileges
+    BOOL process_created = false;
 
-    HWND hwnd = GetShellWindow();
-    if (!hwnd)
+    if (is_process_elevated())
     {
-        goto LExit;
-    }
-    DWORD pid;
-    GetWindowThreadProcessId(hwnd, &pid);
+        HWND hwnd = GetShellWindow();
+        if (!hwnd)
+        {
+            goto LFallBack;
+        }
+        DWORD pid;
+        GetWindowThreadProcessId(hwnd, &pid);
 
-    process = OpenProcess(PROCESS_CREATE_PROCESS, FALSE, pid);
-    if (!process)
-    {
-        goto LExit;
-    }
+        process = OpenProcess(PROCESS_CREATE_PROCESS, FALSE, pid);
 
-    InitializeProcThreadAttributeList(nullptr, 1, 0, &size);
-    pptal = (PPROC_THREAD_ATTRIBUTE_LIST) new char[size];
-    if (!pptal)
-    {
-        goto LExit;
-    }
+        // Try to run the Settings process with non-elevated privileges
+        InitializeProcThreadAttributeList(nullptr, 1, 0, &size);
+        pptal = (PPROC_THREAD_ATTRIBUTE_LIST) new char[size];
+        if (!pptal)
```

Comment:
this condition will never be true, since by default new throws instead of returning `nullptr`. We might change a previous line to
```cpp
        pptal = (PPROC_THREAD_ATTRIBUTE_LIST) new(std::nothrow) char[size];
```